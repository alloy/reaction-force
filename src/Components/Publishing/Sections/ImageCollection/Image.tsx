import React from "react"
import { resize } from "../../../../Utils/resizer"
import { GLOBAL_IMAGE_QUALITY } from '../../Constants'
import { Layout, SectionLayout } from "../../Typings"
import { Caption } from "../Caption"
import { ImageWrapper } from "./ImageWrapper"

interface ImageProps extends React.HTMLProps<HTMLDivElement> {
  editCaption?: any
  image?: any
  layout?: Layout
  sectionLayout?: SectionLayout
  width?: number | string
  height?: number | string
}

export const Image: React.SFC<ImageProps> = props => {
  const {
    editCaption,
    height,
    image,
    layout,
    sectionLayout,
    width,
  } = props

  const src = resize(image.url, { width: 1200, quality: GLOBAL_IMAGE_QUALITY })
  const alt = image.caption.replace(/<[^>]*>/g, "") /* strip caption html */

  return (
    <div className="article-image">
      <ImageWrapper
        layout={layout}
        src={src}
        width={width}
        height={height}
        alt={alt}
        index={image.index}
      />
      {editCaption
        ? editCaption()
        : <Caption
          caption={image.caption}
          layout={layout}
          sectionLayout={sectionLayout} />
      }
    </div>
  )
}

Image.defaultProps = {
  width: "100%",
  height: "auto",
}
