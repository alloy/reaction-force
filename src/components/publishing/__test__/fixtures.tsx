export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini = {
  type: "image_set",
  layout: "mini",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const FeatureHeaders = [
  {
    layout: "text",
    title: "Looking Into the Future",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    layout: "split",
    title: "New York's Next Art District",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    layout: "full",
    title: "New York's Next Art District",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]
