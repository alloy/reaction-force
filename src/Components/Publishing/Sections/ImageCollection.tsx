import { find } from "lodash"
import React from "react"
import sizeMe from "react-sizeme"
import styled from "styled-components"
import fillwidthDimensions from "../../../Utils/fillwidth"
import { pMedia } from "../../Helpers"
import { SIZE_ME_REFRESH_RATE } from "../Constants"
import { SectionLayout } from "../Typings"
import { Artwork } from "./Artwork"
import { Image } from "./Image"
import { ImageCollectionItem } from "./ImageCollectionItem"

interface ImageCollectionProps {
  images: any
  targetHeight?: number
  gutter?: number
  sectionLayout?: SectionLayout
  size?: {
    width: number
  }
}

class ImageCollectionComponent extends React.PureComponent<
  ImageCollectionProps,
  null
> {
  static defaultProps = {
    size: {
      width: 680,
    },
  }

  renderImages(dimensions) {
    const { gutter, images, sectionLayout, size: { width } } = this.props

    const renderedImages = images.map((image, i) => {
      const url = image.url || image.image

      let imageSize
      if (width <= 600 || dimensions.length === 1) {
        imageSize = {}
      } else {
        imageSize = find(dimensions, ["__id", url])
      }

      let renderedImage
      if (image.type === "image") {
        renderedImage = (
          <Image
            image={image}
            sectionLayout={sectionLayout}
            width={imageSize.width}
            height={imageSize.height}
          />
        )
      } else if (image.type === "artwork") {
        renderedImage = (
          <Artwork
            artwork={image}
            sectionLayout={sectionLayout}
            width={imageSize.width}
            height={imageSize.height}
          />
        )
      } else {
        return false
      }

      const margin = i === dimensions.length - 1 ? 0 : gutter

      return (
        <ImageCollectionItem key={i} margin={margin} width={imageSize.width}>
          {renderedImage}
        </ImageCollectionItem>
      )
    })
    return renderedImages
  }

  render() {
    const { gutter, images, size, targetHeight } = this.props
    const dimensions = fillwidthDimensions(
      images,
      size.width,
      gutter,
      targetHeight
    )
    const renderedImages = this.renderImages(dimensions)

    return <ImageCollectionContainer>{renderedImages}</ImageCollectionContainer>
  }
}

const ImageCollectionContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;

  ${pMedia.xs`
    flex-direction: column;
  `};
`

const sizeMeOptions = {
  refreshRate: SIZE_ME_REFRESH_RATE,
  noPlaceholder: true,
}

export const ImageCollection = sizeMe(sizeMeOptions)(ImageCollectionComponent)
