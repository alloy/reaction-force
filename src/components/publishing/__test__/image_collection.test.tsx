import * as React from "react"
import * as renderer from "react-test-renderer"

import ImageCollection from "../image_collection"
import { Images } from "./fixtures"

it("renders properly", () => {
  const image = renderer.create(<ImageCollection images={Images} width={900} targetHeight={400} gutter={10} />).toJSON()
  expect(image).toMatchSnapshot()
})
