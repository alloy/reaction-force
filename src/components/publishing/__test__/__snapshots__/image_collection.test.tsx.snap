// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="file__ImageCollectionContainer-s47wg1-0 eXgPCO"
>
  <div
    className="file-s295gfq-0 dQkKFU"
    height={298}
    margin={10}
    width={415}
  >
    <div
      className="display-artwork"
    >
      <img
        className="display-artwork__image"
        src="https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg"
        width="100%"
      />
      <div
        className="display-artwork__caption"
      >
        <div
          className="file__TruncatedLine-s1fags0c-0 cAwELk"
        >
          <strong>
            <span>
              Fernando Botero
              
            </span>
          </strong>
        </div>
        <div
          className="file__TruncatedLine-s1fags0c-0 cAwELk"
        >
          <span
            className="title"
          >
            <em>
              Nude on the Beach
            </em>
          </span>
          <span
            className="spacer"
          >
            , 
          </span>
          <span
            className="date"
          >
            2000
          </span>
        </div>
        <div
          className="file__TruncatedLine-s1fags0c-0 cAwELk"
        >
          Gary Nader
        </div>
      </div>
    </div>
  </div>
  <div
    className="file-s295gfq-0 gkaePh"
    height={299}
    margin={10}
    width={224}
  >
    <div
      className="article-image"
    >
      <img
        className="file__BlockImage-mvwcj2-3 WFCj"
        src="https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg"
        width="100%"
      />
      <div
        className="file__CaptionContainer-mvwcj2-2 dqVbIF"
      >
        <div
          className="file__Figcaption-mvwcj2-0 frYyrr"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Photo by Adam Kuehl for Artsy.</p>",
            }
          }
        />
        <div
          className="file__CaptionLink-mvwcj2-1 bmAGxU"
        >
          View Fullscreen
        </div>
      </div>
    </div>
  </div>
  <div
    className="file-s295gfq-0 etObiv"
    height={299}
    margin={0}
    width={238}
  >
    <div
      className="article-image"
    >
      <img
        className="file__BlockImage-mvwcj2-3 WFCj"
        src="https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg"
        width="100%"
      />
      <div
        className="file__CaptionContainer-mvwcj2-2 dqVbIF"
      >
        <div
          className="file__Figcaption-mvwcj2-0 frYyrr"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
            }
          }
        />
        <div
          className="file__CaptionLink-mvwcj2-1 bmAGxU"
        >
          View Fullscreen
        </div>
      </div>
    </div>
  </div>
</div>
`;
